python/tskit/intervals.py

```diff
@@ -0,0 +1,754 @@
+        header, rows = self._text_header_and_rows(
+            limit=tskit._print_options["max_lines"]
+        )
+        return util.html_table(rows, header=header)
+
+    def __repr__(self):
+        return f"RateMap(position={repr(self.position)}, rate={repr(self.rate)})"
+
+    #
+    # Methods for building rate maps.
+    #
+
+    def copy(self) -> RateMap:
+        """
+        Returns a deep copy of this RateMap.
+        """
+        # We take read-only copies of the arrays in the constructor anyway, so
+        # no need for copying.
+        return RateMap(position=self.position, rate=self.rate)
+
+    def slice(self, left=None, right=None, *, trim=False) -> RateMap:  # noqa: A003
+        """
+        Returns a subset of this rate map in the specified interval.
+
+        :param float left: The left coordinate (inclusive) of the region to keep.
+            If ``None``, defaults to 0.
+        :param float right: The right coordinate (exclusive) of the region to keep.
+            If ``None``, defaults to the sequence length.
+        :param bool trim: If True, remove the flanking regions such that the
+            sequence length of the new rate map is ``right`` - ``left``. If ``False``
+            (default), do not change the coordinate system and mark the flanking
+            regions as "unknown".
+        :return: A new RateMap instance
+        :rtype: RateMap
+        """
+        left = 0 if left is None else left
+        right = self.sequence_length if right is None else right
+        if not (0 <= left < right <= self.sequence_length):
+            raise KeyError(f"Invalid slice: left={left}, right={right}")
+
+        i = self.find_index(left)
+        j = i + np.searchsorted(self.position[i:], right, side="right")
+        if right > self.position[j - 1]:
+            j += 1
+
+        position = self.position[i:j].copy()
+        rate = self.rate[i : j - 1].copy()
+        position[0] = left
+        position[-1] = right
+
+        if trim:
+            # Return trimmed map with changed coords
+            return RateMap(position=position - left, rate=rate)
+
+        # Need to check regions before & after sliced region are filled out:
+        if left != 0:
+            if np.isnan(rate[0]):
+                position[0] = 0  # Extend
+            else:
+                rate = np.insert(rate, 0, np.nan)  # Prepend
+                position = np.insert(position, 0, 0)
+        if right != self.position[-1]:
+            if np.isnan(rate[-1]):
+                position[-1] = self.sequence_length  # Extend
+            else:
+                rate = np.append(rate, np.nan)  # Append
+                position = np.append(position, self.position[-1])
+        return RateMap(position=position, rate=rate)
+
+    @staticmethod
+    def uniform(sequence_length, rate) -> RateMap:
+        """
+        Create a uniform rate map
+        """
+        return RateMap(position=[0, sequence_length], rate=[rate])
```

Comment:
Here too.

Reply:
I just added that as one of the example rate maps, which is clearer anyway